Create v5 docs for Cordova

Sporiff opened the pull request:
Closes https://adjustcom.atlassian.net/browse/ADAPP-18614

Sporiff commented:
@genadyb @shashanksu The preview is available [here](https://dev-docs-git-adapp-18614-cordova-v-5-int-b6b1e7-product-content.vercel.app/en/sdk/cordova?version=v5)

shashanksu commented:
- [Huawei Referrer API ](https://dev-docs-git-adapp-18614-cordova-v-5-int-b6b1e7-product-content.vercel.app/en/sdk/cordova?version=v5#huawei-referrer-api)- In v5 it is added as [plugin](https://dev.adjust.com/en/sdk/android/migration/v4-to-v5#new-plugins).  So same applies here (similar to React Native v5).
- [Meta referrer Integration](https://dev-docs-git-adapp-18614-cordova-v-5-int-b6b1e7-product-content.vercel.app/en/sdk/cordova?version=v5#meta-referrer-integration) - In v5 it is added as [plugin](https://dev.adjust.com/en/sdk/android/migration/v4-to-v5#new-plugins).  So same applies here (similar to React Native v5).

uerceg commented on src/content/docs/sdk/cordova/features/subscriptions.mdx:
````diff
@@ -2,8 +2,351 @@
+<CodeBlock title="Method signature">
+
+```ts
+public addCallbackParameter(key: string, value: string): void
````
```suggestion
addCallbackParameter(key: string, value: string): void
```
☝️ This signature is in TypeScript.

JavaScript:
```js
AdjustPlayStoreSubscription.prototype.addCallbackParameter = function(key, value)
```

uerceg commented on src/content/docs/sdk/cordova/features/subscriptions.mdx:
````diff
@@ -2,8 +2,351 @@
+<CodeBlock title="Method signature">
+
+```ts
+public addPartnerParameter(key: string, value: string): void
````
```suggestion
addPartnerParameter(key: string, value: string): void
```
☝️ This signature is in TypeScript.

JavaScript:
```js
AdjustAppStoreSubscription.prototype.addPartnerParameter = function(key, value)
```

uerceg commented on src/content/docs/sdk/cordova/features/subscriptions.mdx:
````diff
@@ -2,8 +2,351 @@
+<CodeBlock title="Method signature">
+
+```ts
+trackAppStoreSubscription: (subscription: AdjustAppStoreSubscription) => void
````
```suggestion
trackAppStoreSubscription(adjustAppStoreSubscription: AdjustAppStoreSubscription): void
```
☝️ This signature is in TypeScript.

JavaScript:
```js
trackAppStoreSubscription: function(adjustAppStoreSubscription)
```

uerceg commented on src/content/docs/sdk/cordova/features/subscriptions.mdx:
```diff
@@ -2,8 +2,351 @@
+var subscription = new AdjustAppStoreSubscription(
+   price,
+   currency,
+   transactionId,
```
```suggestion
   transactionId
```

uerceg commented on src/content/docs/sdk/cordova/configuration.mdx:
````diff
@@ -3,8 +3,305 @@ title: Configuration
+<CodeBlock title="Method signature">
+
+```js
+setLogLevel(logLevel: AdjustLogLevel)
````
```suggestion
setLogLevel(logLevel: AdjustLogLevel): void
```
☝️ This signature is in TypeScript.

JavaScript:
```js
AdjustConfig.prototype.setLogLevel = function(logLevel)
```

uerceg commented on src/content/docs/sdk/cordova/configuration.mdx:
````diff
@@ -3,8 +3,305 @@ title: Configuration
+<CodeBlock title="Method signature">
+
+```js
+setDefaultTracker(defaultTracker: string)
````
```suggestion
setDefaultTracker(defaultTracker: string): void
```
☝️ This signature is in TypeScript.

JavaScript:
```js
AdjustConfig.prototype.setDefaultTracker = function(defaultTracker)
```

uerceg commented on src/content/docs/sdk/cordova/features/preinstalled.mdx:
```diff
@@ -2,8 +2,59 @@
+
+You can use the Adjust SDK to record activity from apps that came preinstalled on a user's device. This enables you to send information from users who didn't download your app from a campaign.
+
+To enable preinstall measurement, call the `enablePreinstallTracking` method of your config object.
```
This part doesn't have method signature. In case it's needed:

TypeScript:
```ts
enablePreinstallTracking(): void
```
JavaScript:
```js
AdjustConfig.prototype.enablePreinstallTracking = function()
```

uerceg commented on src/content/docs/sdk/cordova/features/preinstalled.mdx:
````diff
@@ -2,8 +2,59 @@
+Adjust.initSdk(adjustConfig);
+```
+
+## Default link token {#default-link-token}
````
This part doesn't contain method signature. In case it's needed:

TypeScript:
```ts
setDefaultTracker(defaultTracker: string): void
```
JavaScript:
```js
AdjustConfig.prototype.setDefaultTracker = function(defaultTracker)
```

uerceg commented on src/content/docs/sdk/cordova/index.mdx:
````diff
@@ -4,8 +4,301 @@ description: Use the Cordova SDK to access Adjust's features in your Cordova app
+   ```js
+   var adjustConfig = new AdjustConfig(
+      "{YourAppToken}",
+      AdjustConfig.EnvironmentSandbox,
````
```suggestion
      AdjustConfig.EnvironmentSandbox
```

uerceg reviewed (commented):
Important point of this review is that we seem to be having method declarations written in TypeScript and not in JavaScript (which is main language of Cordova SDK). I have added JavaScript code snippet next to each TypeScript snippet and I believe we should add those tabs to be able to flip between the two.

Also, every code snippet was mostly in JavaScript. Those are just updated to reflect JavaScriptness. Right now I can't provide functional TypeScript equivalents, but we can work on that in the future and then also maybe have code snippets tabbed so that people can flip between TS and JS.

In case you have any questions or doubts, feel free to ping.

uerceg reviewed (commented):
Important point of this review is that we seem to be having method declarations written in TypeScript and not in JavaScript (which is main language of Cordova SDK). I have added JavaScript code snippet next to each TypeScript snippet and I believe we should add those tabs to be able to flip between the two.

Also, every code snippet was mostly in JavaScript. Those are just updated to reflect JavaScriptness. Right now I can't provide functional TypeScript equivalents, but we can work on that in the future and then also maybe have code snippets tabbed so that people can flip between TS and JS.

In case you have any questions or doubts, feel free to ping.

Sporiff commented:
@uerceg Regarding the use of TS vs. JS, it makes sense to write the method signatures in TS just because it makes it very clear what data types need to be used with each function. I don't think it matters if the method signature is in TS and the code snippets are in JS, does it? As long as it's clear to the user what data types must be used.

uerceg commented:
> @uerceg Regarding the use of TS vs. JS, it makes sense to write the method signatures in TS just because it makes it very clear what data types need to be used with each function. I don't think it matters if the method signature is in TS and the code snippets are in JS, does it? As long as it's clear to the user what data types must be used.

@Sporiff I see and fine by me. Maybe if there's a way just to highlight that method signatures are in TypeScript and code snippets are in JavaScript, just to make things clear and avoid potential copy / paste issues in TypeScript apps?

Sporiff commented:
@uerceg regarding the smaller style changes such as removal of commas/addition of spaces, these are added by our linter automatically as a pre-commit hook. I'm not going to remove them as they'll just get added again at a later date, and the linter at least will keep us consistent across all documentation.

Sporiff commented on src/content/docs/sdk/cordova/features/deep-links.mdx:
```diff
@@ -3,8 +3,172 @@ title: Deep linking
+
+The SDK can read deep link data after a user opens your app from a link.
+
+## Set up deep linking {#set-up-deep-linking}
```
@uerceg I'm going to go with option 2 here. I think we still need to mention that the SDK isn't responsible for configuring deep link handling and this must be implemented separately, but I don't feel comfortable linking to unmaintained plugins.

uerceg replied:
SGTM. 👌

Sporiff commented:
@uerceg I've been through and addressed the points raised in your review. Could you take another look and let me know if anything else stands out?

[Preview](https://dev-docs-git-adapp-18614-cordova-v-5-int-b6b1e7-product-content.vercel.app/en/sdk/cordova?version=v5)

uerceg commented on src/content/docs/sdk/cordova/features/deep-links.mdx:
```diff
@@ -3,8 +3,158 @@ title: Deep linking
+);
+
+adjustConfig.setDeferredDeeplinkCallback(function (deeplink) {
+   console.log("Deferred deep link URL content: " + deeplink.deeplink);
```
```suggestion
   console.log("Deferred deep link URL content: " + deeplink);
```

uerceg commented on src/content/docs/sdk/cordova/features/deep-links.mdx:
````diff
@@ -3,8 +3,172 @@ title: Deep linking
+
+To reattribute your user, you need to call the `Adjust.processDeeplink` method when the app receives deep link content. The Adjust SDK then looks for new attribution data within the deep link. If the SKD finds new information, it forwards the information to Adjust's servers for reattribution.
+
+```js
````
@Sporiff This one still to be addressed, since current preview is still showing the React Native code in the snippet.

Sporiff replied:
@uerceg Updated. Could you take a look at the new content?

uerceg replied:
@Sporiff Added a minor fix to the code snippet, other than that, looking good!

uerceg commented on src/content/docs/sdk/cordova/features/events.mdx:
```diff
@@ -2,8 +2,640 @@
+  environment      sandbox
+  event_count      3
+  event_token      g3mfiw
+  transaction_id   5e85484b-1ebc-4141-aab7-25b869e54c49
```
```suggestion
  deduplication_id   5e85484b-1ebc-4141-aab7-25b869e54c49
```

uerceg reviewed (commented):
@Sporiff Lookin' nice. Few smaller comments + [this PR](https://github.com/adjust/dev-docs/pull/1069) to be checked since it contains v5 stuff.

uerceg commented on src/content/docs/sdk/cordova/features/deep-links.mdx:
```diff
@@ -3,8 +3,176 @@ title: Deep linking
+function handleOpenURL(url) {
+   setTimeout(function () {
+      // Check content of the url object and get information about the URL.
+      Adjust.appWillOpenUrl(url);
```
```suggestion
      Adjust.appWillOpenUrl(new AdjustDeeplink(url));
```

uerceg commented on src/content/docs/sdk/cordova/features/deep-links.mdx:
```diff
@@ -3,8 +3,176 @@ title: Deep linking
+
+   didLaunchAppFromLink: function (eventData) {
+      // Check content of the url object and get information about the URL.
+      Adjust.appWillOpenUrl(eventData.url);
```
```suggestion
      Adjust.appWillOpenUrl(new AdjustDeeplink(eventData.url));
```
